public/src/components/createTrip/createTrip.jsx

```diff
@@ -1,185 +0,0 @@
-import React from 'react';
-import  createTrip  from '../../actions/createTripsActions.js';
-import { connect } from 'react-redux';
-import getLocations from '../../actions/getLocationActions';
-import SelectLocations from './SelectLocations.jsx';
-
-class CreateTrip extends React.Component{
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      tripdate: '',
-      time: '',
-      location_from: '',
-      location_to: '',
-      passing_by:'',
-      pass_point_time:'',
-      available_seats: 0
-    };
-  }
-
-  componentWillMount(){
-    this.props.Locations();
-  }
-
-  changeTripDate(ev) {
```

Comment:
these update callbacks all have the same logic. Why not make one reusable function like 
updateInfo(item){
return (ev) => {
state = this.state;
state[item] = ev.target.value;
this.setState(state);
}. 

Then you can reuse on each element with onChange = {this.updateInfo([ITEM])}. Make sure to bind function on the constructor (this.updateInfo = this.updateInfo.bind(this)